stake_vote/src/bootstrap.rs

```diff
@@ -0,0 +1,404 @@
+use crate::epoch::ScheduleEpochData;
+use crate::leader_schedule::LeaderScheduleGeneratedData;
+use crate::stake::StakeMap;
+use crate::stake::StakeStore;
+use crate::utils::{Takable, TakeResult};
+use crate::vote::EpochVoteStakes;
+use crate::vote::EpochVoteStakesCache;
+use crate::vote::VoteMap;
+use crate::vote::VoteStore;
+use anyhow::bail;
+use futures::future::join_all;
+use futures_util::stream::FuturesUnordered;
+use solana_client::client_error::ClientError;
+use solana_client::rpc_client::RpcClient;
+use solana_client::rpc_response::RpcVoteAccountStatus;
+use solana_lite_rpc_core::stores::data_cache::DataCache;
+use solana_lite_rpc_core::structures::leaderschedule::CalculatedSchedule;
+use solana_lite_rpc_core::structures::leaderschedule::LeaderScheduleData;
+use solana_sdk::account::Account;
+use solana_sdk::commitment_config::CommitmentConfig;
+use solana_sdk::pubkey::Pubkey;
+use std::time::Duration;
+use tokio::task::JoinHandle;
+
+//File where the Vote and stake use to calculate the leader schedule at epoch are stored.
+//Use to bootstrap current and next epoch leader schedule.
+//TODO to be removed with inter RPC bootstrap and snapshot read.
+pub const CURRENT_EPOCH_VOTE_STAKES_FILE: &str = "current_vote_stakes.json";
+pub const NEXT_EPOCH_VOTE_STAKES_FILE: &str = "next_vote_stakes.json";
+
+pub async fn bootstrap_scheduleepoch_data(data_cache: &DataCache) -> ScheduleEpochData {
+    let new_rate_activation_epoch = solana_sdk::feature_set::FeatureSet::default()
+        .new_warmup_cooldown_rate_epoch(data_cache.epoch_data.get_epoch_schedule());
+
+    let bootstrap_epoch = crate::utils::get_current_epoch(data_cache).await;
+    ScheduleEpochData::new(
+        bootstrap_epoch.epoch,
+        bootstrap_epoch.slots_in_epoch,
+        data_cache
+            .epoch_data
+            .get_last_slot_in_epoch(bootstrap_epoch.epoch),
+        bootstrap_epoch.absolute_slot,
+        new_rate_activation_epoch,
+    )
+}
+
+// Return the current and next epoxh leader schedule and the current epoch stakes of vote accounts
+// if the corresponding files exist.
+pub fn bootstrap_leaderschedule_from_files(
+    slots_in_epoch: u64,
+) -> Option<(CalculatedSchedule, RpcVoteAccountStatus)> {
+    bootstrap_current_leader_schedule(slots_in_epoch)
+        .map(|(leader_schedule, current_epoch_stakes, _)| {
+            let vote_acccounts = crate::vote::get_rpc_vote_account_info_from_current_epoch_stakes(
+                &current_epoch_stakes,
+            );
+            (leader_schedule, vote_acccounts)
+        })
+        .ok()
+}
+
+/*
+Bootstrap state changes
+
+  InitBootstrap
+       |
+  |Fetch accounts|
+
```

Comment:
How do you create these trees ?

Reply:
I use pipe key (Alt Gr 6) on french keyboard.

Reply:
I though you use some library and not make it by hand.